### PR TITLE
docs: Update test specification document for `AccountCreateTransaction`

### LINKED ISSUE #199: Update `AccountCreateTransaction` documentation
Update all the docs for `AccountCreateTransaction` to the specifications and conventions we have worked out in various meetings. When approved and merged, the tests should be updated to match what is described in the docs, as well as naming conventions, etc. `AccountCreateTransaction` should then be used as a model for other request types going forward.

### PR DESCRIPTION
**Pull request by rwalworth**
This PR is the first step towards updating our current TCK tests and documentation to the format we are going to be using going forward for all tested Hedera request types. Before any development is done for any TCK tests, the tests must be fully documented and agreed upon by leads of our various SDKs. This PR will act as the documentation for everyone's agreement and input.

This PR is the first of four steps that will take place for the development of the `AccountCreateTransaction` TCK tests. This process will be the same for all Hedera request types for which TCK tests are written. As a reminder, the process is as follows:

1. Document tests for the request type. The tests should test all APIs associated with the request type, as well as any edge cases or special behaviors. The documentation should use the `test-specifications/testSpecificationsTemplate.md` file as a template. Once reviewed and approved by all designated SDK leads, the test documentation can be merged.
2. The TCK test driver code is written. The approved and merged documentation should be used to discern what tests to write and how they should operate. Once the initial development is complete, a PR should be made but NOT merged until all TCK SDK servers have tested against the branch on which the new TCK code lives.
3. The SDK TCK servers should add methods to support the new TCK tests. Again, using the documentation that has already been approved by all and merged, there should be little question as to what tests are being written and what needs to be supported. Once the initial development is done here, the branch on which the TCK test driver code lives should be used to test against the new SDK TCK server additions. Once all tests work as expected, a PR can be put up for the SDK server work and an approval can be added to the TCK test driver code PR.
4. Once the TCK test driver code has been approved by all SDK leads, all PRs can be merged. It is important that all PRs be merged at the same time to ensure all SDK servers and the TCK test driver stay in parity.

While a good amount of work has been done for `AccountCreateTransaction` already, new guidelines and conventions have been discussed and agreed upon, and this PR implements them on top of what has already been done. There will be succeeding PRs that will update the code to incorporate the new conventions as well; this PR is meant only for the documentation.

In addition to the changes for the `AccountCreateTransaction` documentation, documentation was added for utility functions that the TCK will use. These are functions not specific to any one Hedera request type, but are still used and I felt should be documented somewhere.

I also changed around a bit how the TCK will use the SDK to generate keys and threw all that processing into one function that will allow us to generate more complex keys and aliases.

Please review and feel free to add/mention any other conventions you'd like to put in place, as I believe it would be beneficial to implement them sooner rather than later.

**NOTE:** This PR also brings in the added tests from #14.

**Related issue(s)**:

Fixes #199

### CODE REVIEW

File: test-specifications/cryptoService/accountCreateTransaction.md [hunk 1]
```diff
@@ -0,0 +1,155 @@
+# AccountCreateTransaction - Test specification
+
+## Description:
+This test specification for AccountCreateTransaction is to be one of many for testing the functionality of the Hedera SDKs. The SDK under test will use the language specific JSON-RPC server return responses back to the test driver.
+
+## Design:
+Each test within the test specification is linked to one of the properties within AccountCreateTransaction. Each property is tested with a mix of boundaries. The inputs for each test are a range of valid, minimum, maximum, negative and invalid values for the method. The expected response of a passed test can be a correct error response code or seen as the result of node queries. Success on the consensus node can be obtained by a queries such as AccountInfoQuery or AccountBalanceQuery, and on the mirror node through the rest API. Error codes are obtained from the response code proto files.
```
Comment (SimiHunjan) [inline test-specifications/cryptoService/accountCreateTransaction.md:7]: When you say success on consensus node what do you mean? Like to successfully see the account property was modified. Or do you mean success as in the transaction was successful? I was thinking success as in getting the record or receipt of the transaction.

File: test-specifications/cryptoService/accountCreateTransaction.md [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+
+### Method Name
+
+`createAccount`
+
+### Parameters
+
+| Parameter Name            | Type   | Required/Optional | Description/Notes                                                                                                                           |
+|---------------------------|--------|-------------------|---------------------------------------------------------------------------------------------------------------------------------------------|
+| key                       | string | optional          | DER-encoded hex string representation for private or public keys. Keylists and threshold keys are the hex of the serialized protobuf bytes. |
```
Comment (SimiHunjan) [inline test-specifications/cryptoService/accountCreateTransaction.md:48]: When we say required or optional do we mean this field is required to have the transaction succeed in normal conditions? The key field is required to be set when creating an account otherwise an account will not be created.

File: test-specifications/cryptoService/accountCreateTransaction.md [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+| maxAutoTokenAssociations  | int32  | optional          |                                                                                                                                             |
+| stakedAccountId           | string | optional          |                                                                                                                                             |
+| stakedNodeId              | int64  | optional          |                                                                                                                                             |
+| declineStakingReward      | bool   | optional          |                                                                                                                                             |
+| alias                     | string | optional          | Hex string representation of a serialized protobuf Key of an ED25519 or ECDSAsecp256k1 public key type.                                     |
```
Comment (SimiHunjan) [inline test-specifications/cryptoService/accountCreateTransaction.md:57]: If you are setting the alias on account create you can only set an ECDSAsecp256k1 public key type.

From the HIP:

```
message CryptoCreateTransactionBody {
	...
	/**
     * The bytes to be used as the account's alias. 
     *
     * The bytes must be formatted as the calcluated last 20 bytes of the
     * keccak-256 of the ECDSA primitive key.</li>
     *
     * All other types of keys, including but not limited to ED25519, ThresholdKey, KeyList, ContractID, and
     * delegatable_contract_id, are not supported.
     *
     * At most only one account can ever have a given alias on the network.
     */
	bytes alias = 18;
}

```

> All other types of keys, including but not limited to ED25519, ThresholdKey, KeyList, ContractID, and
>      * delegatable_contract_id, are not supported.

Comment (SimiHunjan) [reply test-specifications/cryptoService/accountCreateTransaction.md:57]: https://hips.hedera.com/hip/hip-583

File: test-specifications/cryptoService/accountCreateTransaction.md [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+| 5       | Creates an account with a valid KeyList of ED25519 and ECDSAsecp256k1 private and public keys | key=<VALID_KEYLIST>                     | The account creation succeeds.                                                 | N                 |
+| 6       | Creates an account with a valid KeyList of nested Keylists (three levels)                     | key=<VALID_NESTED_KEYLIST>              | The account creation succeeds.                                                 | N                 |
+| 7       | Creates an account with no key                                                                |                                         | The account creation fails with a KEY_REQUIRED response code from the network. | N                 |
+| 8       | Creates an account with an invalid key                                                        | key=<INVALID_KEY>                       | The account creation fails with an SDK internal error.                         | N                 |
```
Comment (rwalworth) [inline test-specifications/cryptoService/accountCreateTransaction.md:74]: I debated adding what the JSON will look like for each test, but thought that would be too cumbersome.. maybe just one example for each property/function? Open to thoughts here

File: test-specifications/cryptoService/accountCreateTransaction.md [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+
+### **Alias:**
+
+- The bytes to be used as the account's alias. The bytes must be the serialization of a protobuf Key message for an ED25519/ECDSA_SECP256K1 primitive key type or, if the account is ECDSA_SECP256K1 based it may also be the public address, calculated as the last 20 bytes of the keccak-256 hash of the ECDSA_SECP256K1 primitive key.
+
+| Test no | Name                                                       | Input                                                     | Expected response                                                                      | Implemented (Y/N) |
+|---------|------------------------------------------------------------|-----------------------------------------------------------|----------------------------------------------------------------------------------------|-------------------|
+| 1       | Creates an account with an ED25519 public key alias        | key=<VALID_KEY>, alias=<VALID_ED25519_PUBLIC_KEY>         | The account creation succeeds and the account has the ED25519 public key alias.        | N                 |
```
Comment (SimiHunjan) [inline test-specifications/cryptoService/accountCreateTransaction.md:152]: This case should fail based on the HIP

File: test-specifications/cryptoService/accountCreateTransaction.md (hunk 2 above)
Comment (rwalworth) [reply test-specifications/cryptoService/accountCreateTransaction.md:48]: This is saying that the `key` field is optional in the JSON RPC API. This was made optional to test the case if an `AccountCreateTransaction` were to be submitted without a set key. This doesn't speak to if the key is required by the consensus node or not, simply if the JSON RPC server should require this field every time it receives a `createAccount` method call. 

The idea with this table was for a developer developing their respective SDK's TCK server, they could take a look at this list to see the parameters for the method they need to develop. With `key` being optional, they would develop their method to then not expect to receive a `key` with every `createAccount` method call from the TCK test driver.

File: test-specifications/cryptoService/accountCreateTransaction.md (hunk 3 above)
Comment (rwalworth) [reply test-specifications/cryptoService/accountCreateTransaction.md:57]: Ah forgot to update this, good catch! I'll update it now

File: test-specifications/cryptoService/accountCreateTransaction.md (hunk 2 above)
Comment (SimiHunjan) [reply test-specifications/cryptoService/accountCreateTransaction.md:48]: Ah okay thanks for the explanation that makes sense!!

File: test-specifications/cryptoService/accountCreateTransaction.md (hunk 1 above)
Comment (rwalworth) [reply test-specifications/cryptoService/accountCreateTransaction.md]: I updated the language here to (hopefully) be a bit more clear.

File: test-specifications/cryptoService/accountCreateTransaction.md (hunk 3 above)
Comment (rwalworth) [reply test-specifications/cryptoService/accountCreateTransaction.md]: Fixed

File: test-specifications/cryptoService/accountCreateTransaction.md (hunk 5 above)
Comment (rwalworth) [reply test-specifications/cryptoService/accountCreateTransaction.md]: Updated the tests here

**Review by SimiHunjan — APPROVED** [review]

File: test-specifications/cryptoService/accountCreateTransaction.md [hunk 6]
```diff
@@ -0,0 +1,156 @@
+# AccountCreateTransaction - Test specification
+
+## Description:
+This test specification for AccountCreateTransaction is to be one of many for testing the functionality of the Hedera SDKs. The SDK under test will use the language specific JSON-RPC server return responses back to the test driver.
+
+## Design:
```
Comment (thenswan) [inline test-specifications/cryptoService/accountCreateTransaction.md:6]: > The inputs for each test are a range of valid, minimum, maximum, negative and invalid values for the method.

This is not always the case, so maybe update this sentence or make it always be the case.

File: test-specifications/testSpecificationsTemplate.md [hunk 7]
```diff
@@ -1,43 +1,59 @@
...
+- Description
 
+| Test no | Name | Input | Expected response | Implemented (Y/N) |
+|---------|------|-------|-------------------|-------------------|
+| 1       |      |       |                   |                   |
+| 2       |      |       |                   |                   |
+| 3       |      |       |                   |                   |
 
-### **PropertyName:**
+### **Property/Function Name:**
```
Comment (thenswan) [inline test-specifications/testSpecificationsTemplate.md:50]: Duplicating to show that there can be multiple of them?

File: test-specifications/utility.md [hunk 8]
```diff
@@ -0,0 +1,293 @@
+# Utility JSON RPC Methods
+
+## Description
+The JSON RPC methods mentioned in this file describe additional methods that should be implemented by a TCK server that provide a utility value of some sort that is not specific to one Hedera request type. These methods can involve, but are not limited to, setting up or tearing down a test environment or using the SDK to generate a key pair to be used by the TCK driver.
+
+## Methods
+
+### `setup`
```
Comment (thenswan) [inline test-specifications/utility.md:8]: There was a "setup1" example in the previous version which connect to the testnet, do we still have it?

File: test-specifications/utility.md [hunk 9]
```diff
@@ -0,0 +1,293 @@
...
+          },
+          {}
+        ]
+      }
+    ]
+  }
+}
+```
+
+#### JSON Response Examples
```
Comment (thenswan) [inline test-specifications/utility.md:233]: Maybe it would be good to put response example under the approp. request example.

**Review by thenswan — COMMENTED** [review]
GJ! I have a couple questions.

File: test-specifications/utility.md [hunk 10]
```diff
@@ -0,0 +1,293 @@
...
+| Parameter Name | Type   | Required/Optional | Input/Output | Description/Notes                                                                                                                                                                                                                                                                                                                             |
+|----------------|--------|-------------------|--------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| type           | string | optional          | Input        | The type of Key to generate. If provided, it MUST be one of `ed25519PrivateKey`, `ed25519PublicKey`, `ecdsaSecp256k1PrivateKey`, `ecdsaSecp256k1PublicKey`, `keyList`, or `thresholdKey`. If not provided, the returned key will be of type `ed25519PrivateKey`, `ed25519PublicKey`, `ecdsaSecp256k1PrivateKey`, or `ecdsaSecp256k1PublicKey` |
```
Comment (0xivanov) [inline test-specifications/utility.md:106]: > If not provided, the returned key will be of type `ed25519PrivateKey`, `ed25519PublicKey`, `ecdsaSecp256k1PrivateKey`, or `ecdsaSecp256k1PublicKey`

Why do we need random key type output? Is there a test case for this?

File: test-specifications/cryptoService/accountCreateTransaction.md [hunk 11]
```diff
@@ -0,0 +1,156 @@
...
+|---------|-------------------------------------------------------------------------------------|---------------------------------------------------|-----------------------------------------------------------------------------------------------|-------------------|
+| 1       | Creates an account with an initial balance                                          | key=<VALID_KEY>, initialBalance=100               | The account creation succeeds and the account contains 100 tinybar.                           | N                 |
+| 2       | Creates an account with a negative initial balance                                  | key=<VALID_KEY>, initialBalance=-1                | The account creation fails with an INVALID_INITIAL_BALANCE response code from the network.    | N                 |
+| 3       | Creates an account with an initial balance higher than the operator account balance | key=<VALID_KEY>, initialBalance=1,000,000,000,000 | The account creation fails with an INSUFFICIENT_PAYER_BALANCE response code from the network. | N                 |
```
[the remark this answers is not in the document]
Comment (rwalworth) [reply test-specifications/cryptoService/accountCreateTransaction.md]: Makes sense, fixed

File: test-specifications/cryptoService/accountCreateTransaction.md [hunk 12]
```diff
@@ -0,0 +1,156 @@
...
+
+### **Receiver Signature Required:**
```
[the remark this answers is not in the document]
Comment (rwalworth) [reply test-specifications/cryptoService/accountCreateTransaction.md:142]: Isn't false the default value? Do we need a test case for that?

File: test-specifications/cryptoService/accountCreateTransaction.md [hunk 13]
```diff
@@ -0,0 +1,156 @@
...
+
+| Test no | Name                                                                       | Input                                          | Expected response                                                                                                                | Implemented (Y/N) |
+|---------|----------------------------------------------------------------------------|------------------------------------------------|----------------------------------------------------------------------------------------------------------------------------------|-------------------|
+| 1       | Creates an account with a max token association set to maximum value       | key=<VALID_KEY>, maxAutoTokenAssociations=1000 | The account creation succeeds and the account has 1000 automatic token associations.                                             | N                 |
+| 2       | Creates an account with a max token association set over the maximum value | key=<VALID_KEY>, maxAutoTokenAssociations=1001 | The account creation fails with a REQUESTED_NUM_AUTOMATIC_ASSOCIATIONS_EXCEEDS_ASSOCIATION_LIMIT response code from the network. | N                 |
+
+### **Staked ID:**
```
[the remark this answers is not in the document]
Comment (rwalworth) [reply test-specifications/cryptoService/accountCreateTransaction.md:293]: Number 6 was removed because `stakedNodeId` is now an `int` instead of a `string`, so it can't be empty. Number 7 was removed by mistake and re-added, thanks for catching that

File: test-specifications/cryptoService/accountCreateTransaction.md (hunk 6 above)
Comment (rwalworth) [reply test-specifications/cryptoService/accountCreateTransaction.md:6]: I added more tests to make this sentence valid

File: test-specifications/testSpecificationsTemplate.md (hunk 7 above)
Comment (rwalworth) [reply test-specifications/testSpecificationsTemplate.md:50]: That was the thought, but I have no issue with removing this if you think it's superfluous

File: test-specifications/utility.md (hunk 9 above)
Comment (rwalworth) [reply test-specifications/utility.md]: I like that, makes it easier to follow. Fixed

File: test-specifications/utility.md (hunk 10 above)
Comment (rwalworth) [reply test-specifications/utility.md:106]: I thought this would mostly for developer convenience, if a key is needed and the developer doesn't really care what type it is. Otherwise for every test where a key is required we'll have to specify what kind of key

**Comment by rwalworth** [comment]
Question: does anyone have any preference as to the casing we used for markdown file names and folder names? We currently have a mix of both (`accountCreateTransaction.md` uses camel-case and is located within the `test-specifications/cryptoService`, which uses kebab-case and camel-case).

File: test-specifications/cryptoService/accountCreateTransaction.md (hunk 4 above)
Comment (thenswan) [reply test-specifications/cryptoService/accountCreateTransaction.md:77]: LGTM

File: test-specifications/cryptoService/accountCreateTransaction.md [hunk 14]
```diff
@@ -43,19 +43,21 @@ const transaction = new AccountCreateTransaction()
...
-| stakedNodeId              | int64  | optional          |                                                                                                                                             |
-| declineStakingReward      | bool   | optional          |                                                                                                                                             |
-| alias                     | string | optional          | Hex string representation of the keccak-256 hash of an ECDSAsecp256k1 public key type.                                                      |
-| signerKey                 | string | optional          | DER-encoded hex string representation of an additional private key required to sign.                                                        |
+| Parameter Name            | Type   | Required/Optional | Input/Output | Description/Notes                                                                                                                           |
```
Comment (thenswan) [inline test-specifications/cryptoService/accountCreateTransaction.md:46]: I think its better to have 2 tables: one for input parameters and the other for output.

File: test-specifications/cryptoService/accountCreateTransaction.md (hunk 12 above)
[the remark this answers is not in the document]
Comment (thenswan) [reply test-specifications/cryptoService/accountCreateTransaction.md:142]: I think its about the decision if we should make it verbose or no. I mean adding tests to follow:

> The inputs for each test are a range of valid, minimum, maximum, negative and invalid values for the method.

File: test-specifications/cryptoService/accountCreateTransaction.md (hunk 6 above)
Comment (thenswan) [reply test-specifications/cryptoService/accountCreateTransaction.md:6]: Replied in another conversation on it.

File: test-specifications/testSpecificationsTemplate.md (hunk 7 above)
Comment (thenswan) [reply test-specifications/testSpecificationsTemplate.md:50]: Understood. I think I am ok with it 🙂

File: test-specifications/utility.md (hunk 8 above)
Comment (thenswan) [reply test-specifications/utility.md:8]: Got it. I think it makes sense to add this info to the updated doc then.

File: test-specifications/cryptoService/accountCreateTransaction.md (hunk 12 above)
[the remark this answers is not in the document]
Comment (rwalworth) [reply test-specifications/cryptoService/accountCreateTransaction.md:142]: Yeah I guess it's probably better to be more verbose than not even if it might be a bit overkill. I'll add testing the default values for each property.

[the remark this answers is not in the document]
Comment (rwalworth) [reply test-specifications/cryptoService/accountCreateTransaction.md]: Added test cases for default cases.

File: test-specifications/cryptoService/accountCreateTransaction.md (hunk 6 above)
Comment (rwalworth) [reply test-specifications/cryptoService/accountCreateTransaction.md]: Added test cases for default cases

File: test-specifications/utility.md (hunk 8 above)
Comment (rwalworth) [reply test-specifications/utility.md:8]: Updated with the setup information and assumptions

File: test-specifications/cryptoService/accountCreateTransaction.md (hunk 14 above)
Comment (rwalworth) [reply test-specifications/cryptoService/accountCreateTransaction.md]: Split out parameters into input and output tables

File: test-specifications/crypto-service/accountCreateTransaction.md [hunk 15]
```diff
@@ -0,0 +1,439 @@
...
+```jsx
+new AccountCreateTransaction()
+
+//example
+
+const transaction = new AccountCreateTransaction()
+  .setKey(privateKey.publicKey)
+  .setInitialBalance(new Hbar(1000))
+  ...
+```
```
Comment (rwalworth) [inline test-specifications/crypto-service/accountCreateTransaction.md:36]: Any objections to get rid of this? I don't think it should necessarily be the TCK's responsibility to show how to use the SDK in its documentation; this is what the SDK examples are for.

Comment (SimiHunjan) [reply test-specifications/crypto-service/accountCreateTransaction.md:36]: true. no objections from me

Comment (thenswan) [reply test-specifications/crypto-service/accountCreateTransaction.md:36]: No objections from me as well 👌🏻

File: test-specifications/cryptoService/accountCreateTransaction.md (hunk 14 above)
Comment (thenswan) [reply test-specifications/cryptoService/accountCreateTransaction.md]: Great. Also I think that 'Required/Optional' column can be removed from the table 'Output Parameters'. Wdyt?

File: test-specifications/crypto-service/accountCreateTransaction.md [hunk 16]
```diff
@@ -0,0 +1,439 @@
...
+| 3       | Creates an account with a valid ED25519 private key                                           | key=<VALID_ED25519_PRIVATE_KEY>         | The account creation succeeds.                                                 | N                 |
+| 4       | Creates an account with a valid ECDSAsecp256k1 private key                                    | key=<VALID_ECDSA_SECP256K1_PRIVATE_KEY> | The account creation succeeds.                                                 | N                 |
+| 5       | Creates an account with a valid KeyList of ED25519 and ECDSAsecp256k1 private and public keys | key=<VALID_KEYLIST>                     | The account creation succeeds.                                                 | N                 |
+| 6       | Creates an account with a valid KeyList of nested Keylists (three levels)                     | key=<VALID_NESTED_KEYLIST>              | The account creation succeeds.                                                 | N                 |
```
Comment (thenswan) [inline test-specifications/crypto-service/accountCreateTransaction.md:80]: Maybe add test with INVALID_NESTED_KEYLIST?

File: test-specifications/crypto-service/accountCreateTransaction.md [hunk 17]
```diff
@@ -0,0 +1,439 @@
...
+| 4       | Creates an account with the staked node ID set to a node ID that doesn't exist        | key=<VALID_KEY>, stakedNodeId=123456789                                                      | The account creation fails with an INVALID_STAKING_ID response code from the network.                 | N                 |
+| 5       | Creates an account with the staked account ID set to an empty account ID              | key=<VALID_KEY>, stakedAccountId=""                                                          | The account creation fails with and SDK internal error.                                               | N                 |
+| 6       | Creates an account with the staked account ID set to an invalid node ID               | key=<VALID_KEY>, stakedNodeId=-100                                                           | The account creation fails with an INVALID_STAKING_ID response code from the network.                 | N                 |
```
Comment (thenswan) [inline test-specifications/crypto-service/accountCreateTransaction.md:313]: I think its about an account with the staked node ID (not an account ID) set to an invalid node ID.

Also maybe should be similar test for an account with the staked account ID set to an invalid node ID.

File: test-specifications/crypto-service/accountCreateTransaction.md [hunk 18]
```diff
@@ -0,0 +1,439 @@
...
+  "jsonrpc": "2.0",
+  "id": 99232,
+  "result": {
+    "accountId": "0.0.12345",
+    "status": "SUCCESS"
+  }
+}
+```
+
+### **Max Automatic Token Associations:**
```
Comment (thenswan) [inline test-specifications/crypto-service/accountCreateTransaction.md:263]: I propose to make it to follow 'memo' example.

File: test-specifications/cryptoService/accountCreateTransaction.md (hunk 6 above)
Comment (thenswan) [reply test-specifications/cryptoService/accountCreateTransaction.md]: Looks good, I added couple more comments to tests.

File: test-specifications/crypto-service/accountCreateTransaction.md (hunk 16 above)
Comment (rwalworth) [reply test-specifications/crypto-service/accountCreateTransaction.md:80]: How would this be different than test #8? Both would have the TCK send a string of bytes to the server that the SDK wouldn't be able to make sense of

File: test-specifications/crypto-service/accountCreateTransaction.md (hunk 18 above)
Comment (rwalworth) [reply test-specifications/crypto-service/accountCreateTransaction.md:263]: Could you elaborate a bit more?

File: test-specifications/crypto-service/accountCreateTransaction.md (hunk 17 above)
Comment (rwalworth) [reply test-specifications/crypto-service/accountCreateTransaction.md:313]: Yeah this was a typo, I'll fix it.

File: test-specifications/cryptoService/accountCreateTransaction.md (hunk 14 above)
Comment (rwalworth) [reply test-specifications/cryptoService/accountCreateTransaction.md]: Sure, I suppose I can't think of any case where there would be optional return data.

File: test-specifications/crypto-service/accountCreateTransaction.md (hunk 17 above)
Comment (rwalworth) [reply test-specifications/crypto-service/accountCreateTransaction.md]: Fixed

File: test-specifications/crypto-service/accountCreateTransaction.md (hunk 15 above)
Comment (rwalworth) [reply test-specifications/crypto-service/accountCreateTransaction.md]: Removed

File: test-specifications/cryptoService/accountCreateTransaction.md (hunk 14 above)
Comment (rwalworth) [reply test-specifications/cryptoService/accountCreateTransaction.md]: Removed

File: test-specifications/crypto-service/accountCreateTransaction.md (hunk 16 above)
Comment (thenswan) [reply test-specifications/crypto-service/accountCreateTransaction.md:67]: Checked one more time and you're right, it won't make any sense then

File: test-specifications/crypto-service/accountCreateTransaction.md (hunk 18 above)
Comment (thenswan) [reply test-specifications/crypto-service/accountCreateTransaction.md:250]: Max Automatic Token Associations tests: min, max, invalid.
Memo tests: valid, min, max, invalid.

So I'm just proposing make Max Automatic Token Associations tests more verbose by adding one test with a valid value 🙂

Comment (rwalworth) [reply test-specifications/crypto-service/accountCreateTransaction.md:250]: Understood, added a test

File: test-specifications/crypto-service/accountCreateTransaction.md [hunk 19]
```diff
@@ -254,8 +254,9 @@ https://docs.hedera.com/hedera/sdks-and-apis/rest-api
...
 | 1       | Creates an account with a max token association set to 0                           | key=<VALID_KEY>, maxAutoTokenAssociations=0    | The account creation succeeds and the account has 0 automatic token associations.                                                | N                 |
-| 2       | Creates an account with a max token association that is the maximum value          | key=<VALID_KEY>, maxAutoTokenAssociations=1000 | The account creation succeeds and the account has 1000 automatic token associations.                                             | N                 |
-| 3       | Creates an account with a max token association that is the maximum value plus one | key=<VALID_KEY>, maxAutoTokenAssociations=1001 | The account creation fails with a REQUESTED_NUM_AUTOMATIC_ASSOCIATIONS_EXCEEDS_ASSOCIATION_LIMIT response code from the network. | N                 |
+| 2       | Creates an account with a max token association set to 100                         | key=<VALID_KEY>, maxAutoTokenAssociations=100  | The account creation succeeds and the account has 0 automatic token associations.                                                | N                 |
```
Comment (thenswan) [inline test-specifications/crypto-service/accountCreateTransaction.md]: Should it have 100 automatic token associations (not 0)?
I would also propose same order as in memo (valid, min, max, invalid) 🙏🏻

Comment (rwalworth) [reply test-specifications/crypto-service/accountCreateTransaction.md]: Agreed, updated the order and language a bit

**Review by agadzhalov — APPROVED** [review]
